src/autoschedulers/anderson2021/CMakeLists.txt

```diff
@@ -0,0 +1,159 @@
+##
+# Resources for the autoscheduler library
+##
+
+set(COMMON_DIR "${Halide_SOURCE_DIR}/src/autoschedulers/common")
+
+# weights
+set(WF_CPP baseline.cpp)
+configure_file(baseline.weights baseline.weights COPYONLY)
+add_custom_command(OUTPUT ${WF_CPP}
+                   COMMAND binary2cpp baseline_weights < baseline.weights > ${WF_CPP}
+                   DEPENDS baseline.weights binary2cpp
+                   VERBATIM)
+
+add_library(anderson2021_weights_obj OBJECT ${WF_CPP})
+
+# cost_model, train_cost_model
+add_executable(anderson2021_cost_model.generator cost_model_generator.cpp)
+target_link_libraries(anderson2021_cost_model.generator PRIVATE Halide::Halide Halide::Generator)
+
+add_halide_library(anderson2021_cost_model FROM anderson2021_cost_model.generator
+                   GENERATOR cost_model
+                   FUNCTION_NAME cost_model
+                   TARGETS cmake)
+add_halide_library(anderson2021_train_cost_model FROM anderson2021_cost_model.generator
+                   GENERATOR train_cost_model
+                   FUNCTION_NAME train_cost_model
+                   TARGETS cmake
+                   USE_RUNTIME anderson2021_cost_model.runtime)
+
+## retrain_cost_model
+if (WITH_UTILS)
+    add_executable(anderson2021_retrain_cost_model
+                   DefaultCostModel.cpp
+                   ${COMMON_DIR}/Weights.cpp
+                   retrain_cost_model.cpp
+                   $<TARGET_OBJECTS:anderson2021_weights_obj>)
+    target_include_directories(anderson2021_retrain_cost_model PRIVATE "${Halide_SOURCE_DIR}/src/autoschedulers/anderson2021")
+    target_link_libraries(anderson2021_retrain_cost_model PRIVATE ASLog anderson2021_cost_model
+        anderson2021_train_cost_model Halide::Halide Halide::Plugin)
+endif ()
+
+###
+## Main autoscheduler library
+###
+
+add_autoscheduler(
+    NAME Anderson2021
+    SOURCES
+    AutoSchedule.cpp
+    DefaultCostModel.cpp
+    FunctionDAG.cpp
+    GPULoopInfo.cpp
+    LoopNest.cpp
+    SearchSpace.cpp
+    State.cpp
+    Tiling.cpp
+    ${COMMON_DIR}/Weights.cpp
+    $<TARGET_OBJECTS:anderson2021_weights_obj>
+)
+
+target_include_directories(Halide_Anderson2021 PUBLIC "${Halide_SOURCE_DIR}/src/autoschedulers/anderson2021")
```

Comment:
I'm very interested in this work and have started to play around, so let me share some findings.
You might be aware already, but, in order to make Halide::Anderson2021 available in installed Halide package, you would need to add it to packaging [here](https://github.com/halide/Halide/blob/147ff481ee5495c21253db58d480e411cfb5b9f1/packaging/CMakeLists.txt#L35). In addition, I needed to modify `PUBLIC` to `PRIVATE` in this line to fix CMake error.